src/libfsm/determinise.c

```diff
@@ -218,6 +219,15 @@ state_closure(struct set *mappings, struct fsm *dfa, const struct fsm_state *nfa
 		return NULL;
 	}
 
+	if (!includeself) {
+		set_remove(&ec, (void *) nfastate);
+
+		if (set_count(ec) == 0) {
+			set_free(ec);
+			ec = NULL;
+		}
+	}
+
 	if (ec == NULL) {
 		return NULL;
 	}
```

Comment:
So, given that you call state_closure a lot, and only remove a state once (at the beginning) under very special conditions, does it make sense to pass this as a boolean into the state_closure function?  I'd be tempted to either write a special remove function or to inline the removal.

Reply:
I did wonder that. There's be no other consumer for `state_closure()` with its own state included, so I'd just delete it. And then I got unhappy about losing information. So not today - perhaps when dredging some of this stuff out to a more general interface.